### ISSUE TITLE
Suggestion: add show for options

### ISSUE THREAD

**Issue by briochemc** [JuliaNLSolvers/Optim.jl#702]
Maybe I just did not know how to find it, but I thought it would be convenient to have a `show` function (or a `print` one?), in order to be able to do something that looks like:
```julia
julia> show(my_options)
    x_tol                      1e-10
    f_tol                      0.0
    g_tol                      1e-8
    outer_x_tol                0.0
    outer_f_tol                0.0
    outer_g_tol                1e-8
    f_calls_limit              0
    g_calls_limit              0
    h_calls_limit              0
    allow_f_increases          false
    allow_outer_f_increases    false
    successive_f_tol           0
    iterations                 1_000
    outer_iterations           1000
    store_trace                false
    show_trace                 false
    extended_trace             false
    show_every                 1
    callback                   nothing
    time_limit                 NaN
```
where `my_options = Optim.Options(x_tol=1e-10)`

I was looking for that sort of option display because the current one is a bit hard to parse for a human:
```julia
julia> show(my_options)
Optim.Options{Float64,Nothing}(1.0e-10, 0.0, 1.0e-8, 0.0, 0.0, 1.0e-8, 0, 0, 0, false, false, 0, 1000, 1000, false, false, false, 1, nothing, NaN)
```

**Comment by pkofod** [comment]
That would be awesome. Is it something you'd maybe want to add? I guess it can just loop over the fieldnames and values to automatically produce this, right?

I'm not too familiar with what methods to overload to make this printing work automatically, say 
```
julia> my_options
```
would just print it, instead of having to do `show` or similar.
